src/Http/Controllers/QuestionnaireApiController.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace EscolaLms\Questionnaire\Http\Controllers;
+
+use EscolaLms\Core\Http\Controllers\EscolaLmsBaseController;
+use EscolaLms\Questionnaire\Http\Controllers\Contracts\QuestionnaireApiContract;
+use EscolaLms\Questionnaire\Http\Requests\QuestionnaireFrontListingRequest;
+use EscolaLms\Questionnaire\Http\Requests\QuestionnaireFrontReadRequest;
+use EscolaLms\Questionnaire\Http\Resources\QuestionnaireResource;
+use EscolaLms\Questionnaire\Repository\Contracts\QuestionnaireRepositoryContract;
+use Illuminate\Contracts\Support\Renderable;
+use Illuminate\Http\JsonResponse;
+
+class QuestionnaireApiController extends EscolaLmsBaseController implements QuestionnaireApiContract
+{
+    private QuestionnaireRepositoryContract $questionnaireRepository;
+
+    public function __construct(QuestionnaireRepositoryContract $questionnaireRepository)
+    {
+        $this->questionnaireRepository = $questionnaireRepository;
+    }
+
+    public function list(QuestionnaireFrontListingRequest $request): JsonResponse
+    {
+        try {
+            $questionnaires = $this->questionnaireRepository->searchAndPaginate(['active' => true]);
+            return $this->sendResponseForResource(QuestionnaireResource::collection($questionnaires), "questionnaire list retrieved successfully");
+        } catch (Renderable $e) {
+            return $this->sendError($e->getMessage());
+        }
+    }
+
+    public function read(QuestionnaireFrontReadRequest $request): JsonResponse
+    {
+        try {
+            $id = $request->getParamId();
+            $questionnaire = $this->questionnaireRepository->find($id);
+            if ($questionnaire && $questionnaire->exists) {
+                if (!$questionnaire->active) {
+                    return $this->sendError(sprintf("You don't have access to questionnaire with id '%s'", $id), 403);
```

Comment:
Shouldn't we use Laravel string helpers for messages? Something like `__("You don't have access to questionnaire with id :id", ['id' => $id])` or `trans("You don't have access to questionnaire with id :id", ['id' => $id])`?

And also many of these `if` statemens in all your Controllers should probably be either moved to corresponding Policy (checking in `view` or `read` policy if something is `active`) or written as Request rules (checking for existence, etc.)

Reply:
done